### ISSUE TITLE
Jenkins Pipepline syntax

### ISSUE THREAD

**Issue by sschueller** [jenkinsci/zulip-plugin#7]
How do I send a message to Zulip in a Jenkins Pipeline (`Jenkinsfile`) ? For example this is how I would do it with hipchat (https://github.com/jenkinsci/hipchat-plugin):

```
pipeline {
  agent any
  stages {
    stage('Notify Start') {
      steps {
        hipchatSend(color: 'GRAY', message: "<a href=\"${RUN_DISPLAY_URL}\">Web Job ${env.BUILD_NUMBER} on ${env.BRANCH_NAME} started</a>")
      }
    }
    stage('Automated Tests') {
      steps {
        sh '''# my tests
            '''
        junit 'tests/_output/run'
      }
      post {
        unstable {
          hipchatSend(color: 'RED', notify: true, message: "<a href=\"${RUN_DISPLAY_URL}\">Some tests in web job ${env.BUILD_NUMBER} on ${env.BRANCH_NAME} failed</a>")

        }
        success {
          hipchatSend(color: 'GREEN', notify: false, message: "<a href=\"${RUN_DISPLAY_URL}\">All tests in web job ${env.BUILD_NUMBER} on ${env.BRANCH_NAME} passed</a>")
        }
      }
    }
    stage('Build and Release') {
      when {
        tag 'release_*'
      }
      post {
        failure {
          hipchatSend(color: 'RED', notify: true, message: "<a href=\"${RUN_DISPLAY_URL}\">Build web job ${env.BUILD_NUMBER} on ${env.BRANCH_NAME} failed</a>")

        }

      }
      steps {
        sh "# build"
      } 
    }
    stage('Notify Deployment') {
      when {
        tag 'release_*'
      }
      steps {
        hipchatSend(color: 'GREEN', notify: false, message: "Web Job ${env.BRANCH_NAME} deployed")
      }
    }
  }
}
```

I also noticed that the Zulip plugin is not showing as an option in the blueocean pipeline editor so maybe this is not supported?

**Comment by timabbott** [comment]
@sschueller I suspect this is not supported with the current plugin.

**Comment by timabbott** [comment]
(Let me see about getting someone to work on improving this)

**Comment by timabbott** [comment]
@sschueller I believe https://github.com/zulip/zulip-jenkins-plugin/pull/8 should have resolved at least parts of this; can you take a look?

**Comment by timabbott** [comment]
@butchyyyy confirmed that this is fixed via #8, so closing.  Thanks again for the report!
